mixins/auth-config.js

```diff
@@ -115,10 +137,7 @@ export default {
             if (!this.model.accessMode) {
               this.model.accessMode = 'unrestricted';
             }
-            await this.model.doAction('testAndApply', {
-              code:   obj.code,
-              config: obj
-            }, { redirectUnauthorized: false });
+            await this.model.doAction('testAndApply', obj, { redirectUnauthorized: false });
```

Comment:
Setting up github initially works fine, however when editing the provider with the same details it errors with `server error while authenticating: Received Error from github bad_verification_code, description from github The code passed is incorrect or expired.` Might be caused here by the `code` being dropped.

Reply:
I don't remember my reasoning for adding `&& !wasEnabled` [here](https://github.com/rancher/dashboard/pull/2309/files/18ed3c9b87856ccc9d62a3f35e6d1d381181799d#diff-b3838ef0862d5e9f887dcd18548f5233e0453b47455b8963fde00b4b5289169dR122) but it was bad. Removing this check allows the oauth config flow to re-run on edit/save.

Reply:
This change here is what was causing the `server error while authenticating: parameter 'activeDirectoryEndpoint' cannot be empty` error.

The backend seems to expect the config to be nested under config and not flat at least for azuread.

Reply:
With my correction to line 121 this morning from `if (configType === 'oauth' && !wasEnabled)` -> `if (configType === 'oauth')` this line shouldn't ever be hit by enabling/editing azureAD, as it needs to follow the oauth route (`auth/test` vuex action).
I'm a little surprised `testAndApply` is an available action for this provider actually, since (assuming I read your code correctly yesterday) it's not needed to enable auth.

Reply:
testAndApply is what actually enables auth for azureAD. I tested with the code you updated this morning and modified this line to fix it.

Reply:
Hmm I misread my own code how supremely embarrassing. Well, modifying that call to testAndApply would interfere github, google, and ldap so I altered azuredad.vue `toSave` to nest the model under a  `config` key (github does the same though the key is `githubConfig`)

Reply:
Tested editing both github and saml, both successfully saved